### PR TITLE
fix(historical): use matching question for historicalanswer

### PR DESCRIPTION
**Pull request by sbor23**
Use correct historical question for finding answer type.

### CODE REVIEW

**Comment by sbor23** [comment]
I'm just leaving this here...
https://github.com/treyhunner/django-simple-history/issues/397

**Comment by open-dynaMIX** [comment]
> I'm just leaving this here...
> [treyhunner/django-simple-history#397](https://github.com/treyhunner/django-simple-history/issues/397)

https://github.com/projectcaluma/caluma/blob/master/caluma/caluma_form/historical_schema.py#L33 ;)

**Comment by open-dynaMIX** [comment]
Will have a closer look next week.

**Comment by sbor23** [comment]
> > I'm just leaving this here...
> > [treyhunner/django-simple-history#397](https://github.com/treyhunner/django-simple-history/issues/397)
> 
> https://github.com/projectcaluma/caluma/blob/master/caluma/caluma_form/historical_schema.py#L33 ;)

Hahaha how did I not realize this :D

File: caluma/caluma_form/tests/test_history.py [hunk 1]
```diff
@@ -372,3 +372,131 @@ def test_historical_table_answer(
     result = schema_executor(historical_query, variables=variables)
     assert not result.errors
     snapshot.assert_match(result.data)
+
+
+def test_history_question_answer(
```
Comment (winged) [inline caluma/caluma_form/tests/test_history.py:377]: Can you add a comment on what aspect is being tested here?

File: caluma/caluma_form/schema.py [hunk 2]
```diff
@@ -963,3 +933,34 @@ def resolve_all_format_validators(self, info):
...
+    models.Question.TYPE_CHOICE: StringAnswer,
+    models.Question.TYPE_TEXTAREA: StringAnswer,
+    models.Question.TYPE_TEXT: StringAnswer,
+    models.Question.TYPE_TABLE: TableAnswer,
+    models.Question.TYPE_FILE: FileAnswer,
+    models.Question.TYPE_DYNAMIC_CHOICE: StringAnswer,
+    models.Question.TYPE_DYNAMIC_MULTIPLE_CHOICE: ListAnswer,
+}
+
+QUESTION_OBJECT_TYPE = {
```
Comment (winged) [inline caluma/caluma_form/schema.py:952]: Good work on refactoring out those lists! However I'd prefer this being plural named (`QUESTION_OBJECT_TYPES`)

File: caluma/caluma_form/tests/test_history.py (hunk 1 above)
Comment (sbor23) [reply caluma/caluma_form/tests/test_history.py]: Gave it a better name instead

Comment (winged) [reply caluma/caluma_form/tests/test_history.py]: I'd still like something along the lines of "test that historical question types retain the right type even when the question type for a given slug changes" or something similar. So you know what the intention is behind the test (I know I should do that more as well...)

Comment (sbor23) [reply caluma/caluma_form/tests/test_history.py]: Did that

**Pull request merged**